File: Medium/3115_Maximum_Prime_Difference/Solution.py
from typing import List, Optional


class Solution:
    def isPrime(self, n: int) -> bool:
        if n <= 1:
            return False
        
        for i in range(2, int(n**0.5) + 1):
            if n % i == 0:
                return False
        
        return True
    
    def maximumPrimeDifference(self, nums: List[int]) -> int:
        left_prime: Optional[int] = None
        right_prime: Optional[int] = None
        
        for i in range(len(nums)):
            if self.isPrime(nums[i]):
                left_prime = i
                break
        
        for i in range(len(nums) - 1, -1, -1):
            if self.isPrime(nums[i]):
                right_prime = i
                break
        
        
        return right_prime - left_prime


if __name__ == "__main__":
    solution = Solution()
    
    # test cases 1
    nums1 = [4, 2, 9, 5, 3]
    print(solution.maximumPrimeDifference(nums1))  # Output: 3

    # test cases 2
    nums2 = [4, 8, 2, 8]
    print(solution.maximumPrimeDifference(nums2))  # Output: 0
        
        

File: Medium/3115_Maximum_Prime_Difference/Solution.java


public class Solution {
    Boolean isPrime(int n) {
        if (n <= 1) return false;

        for (int i = 2; i * i <= n; ++i) {
            if (n % i == 0) return false;
        }

        return true;
    }

    int maximumPrimeDifference(int[] nums) {
        
        int left_prime = -1;
        int right_prime = -1;

        for (int i = 0; i < nums.length; i++) {
            if (isPrime(nums[i])) {
                left_prime = i;
                break;
            }
        }

        for (int i = nums.length - 1; i >= 0; i--) {
            if (isPrime(nums[i])) {
                right_prime = i;
                break;
            }
        }
        
        return right_prime - left_prime;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        
        // test cases 1
        int[] nums1 = {4,2,9,5,3};
        System.out.println(solution.maximumPrimeDifference(nums1)); // Output: 3

        // test cases 2
        int[] nums2 = {4,8,2,8};
        System.out.println(solution.maximumPrimeDifference(nums2)); // Output: 0
        
    }
}